builders/build/canonical-contracts/precompiles/batch.md

```diff
@@ -0,0 +1,225 @@
+
+ --8<-- 'text/batch/simple-message-call-data.md'
+
+The calldata can be broken into 5 lines, where the first line is the function selector and the next four are the input data. 
+
+Note that the second line is equal to 1, which is the **id** that was provided. 
+
+What's left has to do with the **message** input. These last three lines are tricky, since strings are encoded with [RLP encoding](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/){target=_blank}. For sake of simplicity, understand that the third line refers to an offset, the fourth line refers to the string's length, and the fifth line is "moonbeam" in hexadecimal format with zeros for padding.
+
+### Function Interaction via Precompile {: #function-interaction-via-precompile }
+
+Interacting with a function is very similar to [sending a native currency](#send-native-currency-via-precompile), since they are both transactions. However, call data is required to properly provide input to functions and a sender may desire to limit the amount of gas spent in each subtransaction. 
+
+The *call_data* and *gas_limit* transactions more relevant for subtransactions that interact with contracts. For each function in the Batch interface, the *call_data* input is an array that corresponds to the call data for each subtransaction. If its length is less than the *to* input, the remaining subtransactions will have no call data. The *gas_limit* input is an array that corresponds to the amount of gas that each can spend for each subtransaction. If its length is less than the *to* input, the remaining transactions will have all of the batch transaction's remaining gas forwarded.
+
+Try using the precompile to send a transaction:
+
+1. Copy the SimpleMessage contract's address with the copy button on the right side of its header. Be sure to also have the [call data from the previous section](#finding-a-contract-interactions-call-data) 
+2. Expand the Batch contract under **Deployed Contracts**
+3. Expand the **batchAll** function
+4. For the *to* input, insert your addresses in the following format: `["0xSIMPLEMESSAGEADDRESS"]`, where 0xSIMPLEMESSAGEADDRESS is the address of the SimpleMessage contract that you previously copied
+5. For the value input, since SimpleMessage does not require any native currency to be paid to it, insert `["0"]` for 0 Wei
+6. For the *call_data* input, insert your call data in the following format: `["CALL_DATA"]`, where CALL_DATA is the call data that you found in the previous section
+7. For the *gas_limit* input, insert `[]`. You can put in a gas limit value, but it is optional 
+8. Press **transact**
+9. Press **Confirm** in the Metamask extension to confirm the transaction
+
+![Batch Function Interaction](/images/builders/build/canonical-contracts/precompiles/batch/batch-6.png)
+
+If you used the same call data as the tutorial, let's check to make sure that an actual change has been made:
```

Comment:
We avoid 2nd person terms such as 'let's' - You can do, `...take the following steps to ensure that all transactions went through:`